### PR TITLE
Correct mapping and migrations for tests to run

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/StudyImpl.java [hunk 1]
```diff
@@ -100,6 +101,7 @@
   private String studyType;
   @Column(name = "alias")
   private String alias;
+  @OneToMany(targetEntity = StudyChangeLog.class)
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/StudyImpl.java]: `mappedBy = "study"`

File: sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql [hunk 2]
```diff
@@ -193,7 +196,12 @@ DROP TABLE PrintService;
 DROP TABLE PrintJob;
 
 
-ALTER TABLE BoxChangeLog ADD boxChangeLogId bigint(20) PRIMARY KEY AUTO_INCREMENT FIRST;
+ALTER TABLE BoxChangeLog ADD boxChangeLogId bigint(20) AUTO_INCREMENT
+-- StartNoTest
+ FIRST
+-- EndNoTest
+;
```
Comment (callunity) [inline sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql]: Fails to migrate: 
```SQL State  : 42000
[ERROR] Error Code : 1075
[ERROR] Message    : Incorrect table definition; there can be only one auto column and it must be defined as a key
[ERROR] Location   : db/migration/V8900__hibernatize_everything.sql (/home/harmstrong/workspace/miso-lims/sqlstore/target/classes/db/migration/V8900__hibernatize_everything.sql)
[ERROR] Line       : 199
[ERROR] Statement  : ALTER TABLE BoxChangeLog ADD boxChangeLogId bigint(20) AUTO_INCREMENT
[ERROR] -- StartNoTest
[ERROR] FIRST
[ERROR] -- EndNoTest
[ERROR] -> [Help 1]
```
(The ones below will also have the same issue)

File: sqlstore/src/test/resources/db/test_migration/V9000__miso_test_data.test.sql [hunk 3]
```diff
@@ -494,10 +501,10 @@ VALUES
 ('2','1', '2', 'box2', 'box2alias', 'box2', 'barcode2','identifcationbarcode2', '1', '1');
 
 DELETE FROM BoxPosition;
-INSERT INTO `BoxPosition` (`BoxPositionId`, `boxId`, `column`, `row`, `lastModifier`)
+INSERT INTO `BoxPosition` (`boxId`, `position`, `targetType`, `targetId`)
 VALUES
-('1', '1', '1', '1', '1'),
-('2', '2', '1', '2', '1');
+('1', 'A1', 'S', 1),
+('2', 'A2', 'S', 2);
```
Comment (callunity) [inline sqlstore/src/test/resources/db/test_migration/V9000__miso_test_data.test.sql]: These should be represented in the UI as `A01`, `C09`, `D10`, etc.

File: sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql (hunk 2 above)
Comment (apmasell) [inline sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql]: I was so focus on H2, I forgot about mySQL's needs. :-(

**Comment by apmasell** [comment]
Fixed.

File: sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql [hunk 4]
```diff
@@ -193,7 +196,12 @@ DROP TABLE PrintService;
 DROP TABLE PrintJob;
 
 
-ALTER TABLE BoxChangeLog ADD boxChangeLogId bigint(20) PRIMARY KEY AUTO_INCREMENT FIRST;
+ALTER TABLE BoxChangeLog ADD boxChangeLogId bigint(20) AUTO_INCREMENT
+-- StartNoTest
+ PRIMARY KEY FIRST
+-- EndNoTest
+;
+ALTER TABLE BoxChangeLog ADD PRIMARY KEY(`boxChangeLogId`);
```
Comment (callunity) [inline sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql]:
``` SQL State  : 42000
[ERROR] Error Code : 1068
[ERROR] Message    : Multiple primary key defined
[ERROR] Location   : db/migration/V8900__hibernatize_everything.sql (/home/harmstrong/workspace/miso-lims/sqlstore/target/classes/db/migration/V8900__hibernatize_everything.sql)
[ERROR] Line       : 204
[ERROR] Statement  : ALTER TABLE BoxChangeLog ADD PRIMARY KEY(`boxChangeLogId`)
[ERROR] -> [Help 1]
```

**Comment by apmasell** [comment]
Fixed. I tested it with the prod DB, which works, and the staging DB, which doesn't due to bad data in the `BoxPosition` table.

**Pull request merged**